net/transaction.go

```diff
@@ -0,0 +1,104 @@
+package net
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	libp2pnetwork "github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+
+	p2pnet "github.com/athanorlabs/go-p2p-net"
+	"github.com/athanorlabs/go-relayer/common"
+)
+
+const (
+	transactionID      = "/transaction/0"
+	transactionTimeout = time.Second * 5
+)
+
+func (h *Host) handleTransactionStream(stream libp2pnetwork.Stream) {
+	msg, err := readStreamMessage(stream, maxMessageSize)
+	if err != nil {
+		log.Debugf("error reading TransactionRequest: %w", err)
+	}
+
+	req, ok := msg.(*TransactionRequest)
+	if !ok {
+		return
```

Comment:
Is this something that happens normally, or is it something abnormal that we might want to log?

Reply:
this can happen if the remote side sends the wrong type of message over the stream. shouldn't happen normally. can add a debug log if you want